packages/react/select/src/Select.tsx

```diff
@@ -262,8 +270,15 @@ const SelectTrigger = React.forwardRef<SelectTriggerElement, SelectTriggerProps>
             // because we are preventing default in `onPointerDown` so effectively
             // this only runs for a label "click"
             event.currentTarget.focus();
+
+            // Open on click when using a touch or pen device
+            if (pointerTypeRef.current !== 'mouse') {
+              handleOpen(event);
+            }
```

Comment:
For posterity, about pen devices in this PR:
- Using an Apple Pencil on iOS yields `pointerType: 'pen'`, but behaves exactly like touch (e.g. opening on pointerdown when intending to scroll the page)
- Using a Wacom tablets on macOS also yields `pointerType: 'pen'`, but behaves exactly like mouse (which don't need pointerdown to scroll, so the current behaviour isn't a problem in this case).

Since we can disambiguate the two, I'm being a bit more defensive here and assuming `pen` is equivalent to `touch`